include/board.c

```diff
@@ -1,33 +1,91 @@
-#include<stdio.h>
+#include <stdio.h>
 #include "board.h"
 
-void init_board(struct Board *board) {
-    for (int i = 0; i < SIZE; i++) {
-        for (int j = 0; j < SIZE; j++) {
+void init_board(struct Board *board)
+{
+    for (int i = 0; i < SIZE; i++)
+    {
+        for (int j = 0; j < SIZE; j++)
+        {
             board->board[i][j] = EMPTY;
         }
     }
 }
 
-
-void print_board(struct Board *board) {
+void print_board(struct Board *board)
+{
     printf("\n");
     printf("    ");
-    for (int col = 0; col < SIZE; col++) {
+    for (int col = 0; col < SIZE; col++)
+    {
         printf("%d ", col);
     }
     printf("\n");
     printf("    ");
-    for (int col = 0; col < SIZE; col++) {
+    for (int col = 0; col < SIZE; col++)
+    {
         printf("--");
     }
     printf("\n");
-    for (int row = 0; row < SIZE; row++) {
+    for (int row = 0; row < SIZE; row++)
+    {
         printf("%d | ", row);
-        for (int col = 0; col < SIZE; col++) {
+        for (int col = 0; col < SIZE; col++)
+        {
             printf("%c ", board->board[row][col]);
         }
         printf("\n");
     }
     printf("\n");
-}
\ No newline at end of file
+}
+
+int check_draw(struct Board *board) // 0: no draw, 1: draw
+{
+    for (int row = 0; row < 3; row++)
+    {
+        for (int col = 0; col < 3; col++)
+        {
+            if (board->board[row][col] != 'X' && board->board[row][col] != 'O')
+            {
+                return 0;
+            }
+        }
```

Comment:
Isnt this the opposite of what you want to do? Explain this part of code

Reply:
int check_valid_move(char board[3][3], int row, int col)
{
   return (board[row][col] == 'X' || board[row][col] == 'O') ? 0 : 1;

}

i think  upper  function is right

same as ->
int check_valid_move(char board[3][3], int row, int col)
{
  
 if (board->board[row][col] != 'X' && board->board[row][col] != 'O')
            {
                return 1;
            }
      return 0;
}

Reply:
> Isnt this the opposite of what you want to do? Explain this part of code

it is checking if at least one index of board is not filled with either X or O then return 0 (condition for not draw)and return 1 if all the index contain either X or O

Reply:
Ah ok, understood